crates/prover/src/runner.rs

```diff
@@ -786,6 +800,55 @@ where
     pub fn get_state_root(&self) -> &Stf::StateRoot {
         &self.state_root
     }
+
+    fn break_sequencer_commitments_into_groups(
+        &self,
+        sequencer_commitments: Vec<SequencerCommitment>,
+    ) -> anyhow::Result<Vec<Vec<SequencerCommitment>>> {
+        let mut result = vec![];
+
+        let mut group = vec![];
+        let mut cumulative_state_diff = StateDiff::new();
+        for sequencer_commitment in sequencer_commitments {
+            group.push(sequencer_commitment.clone());
+            for l2_height in sequencer_commitment.l2_start_block_number
+                ..=sequencer_commitment.l2_end_block_number
+            {
+                let state_diff = self
+                    .ledger_db
+                    .get_l2_state_diff(BatchNumber(l2_height))?
+                    .ok_or(anyhow!(
+                        "Could not find state diff for L2 range {}-{}",
+                        sequencer_commitment.l2_start_block_number,
+                        sequencer_commitment.l2_end_block_number
+                    ))?;
+                cumulative_state_diff = merge_state_diffs(cumulative_state_diff, state_diff);
+            }
+
+            let serialized_state_diff = bincode::serialize(&cumulative_state_diff)?;
+
+            // The threshold is reached when we hit 95% of the limit to prevent exceeding it.
+            let state_diff_threshold_reached = serialized_state_diff.len() as u64
+                > (MAX_STATEDIFF_SIZE_PROOF_THRESHOLD
+                    .saturating_mul(95)
+                    .saturating_div(100));
+
```

Comment:
instead checking this way, we should check if the addition of nth commitment makes the merged commitment go over the limit

in that case we group until n-1.

in the case that a single commitment by itself goes over the limit, we have to accept that and have a single commitment group. (and that's solved by having [chainwayxyz/citrea#744](https://github.com/chainwayxyz/citrea/issues/744))